core/src/OpenACC/Kokkos_OpenACC_ParallelScan_Range.hpp

```diff
@@ -0,0 +1,215 @@
+/*
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 3.0
+//       Copyright (2020) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef KOKKOS_OPENACC_PARALLEL_SCAN_RANGE_HPP
+#define KOKKOS_OPENACC_PARALLEL_SCAN_RANGE_HPP
+
+#include <OpenACC/Kokkos_OpenACC.hpp>
+#include <OpenACC/Kokkos_OpenACC_FunctorAdapter.hpp>
+#include <Kokkos_Parallel.hpp>
+
+namespace Kokkos::Experimental::Impl {
+template <class IndexType, class Functor, class ValueType>
+void OpenACCParallelScanRangePolicy(IndexType begin, IndexType end,
+                                    Functor afunctor, ValueType init_value,
+                                    int async_arg) {
+  auto const functor(afunctor);
+  const IndexType N                  = end - begin;
+  constexpr IndexType chunk_size     = 128;
+  constexpr IndexType log_chunk_size = 7;
+  const IndexType n_chunks           = (N + chunk_size - 1) / chunk_size;
+  const IndexType nteams             = n_chunks > 512 ? 512 : n_chunks;
```

Comment:
Is there any reason for the hard-coded chunk size?  Is it the same as what parallel_reduce/for does?

Reply:
Is it possible to get the default value from m_policy.chunk_size()? See the lines 178--212.

Reply:
We can use m_policy.chunk_size(). It looks that chunk size must me multiple of 64. So in case that chunk_size is not multiple of 64 then we can either i) use the closer multiple of 64 of chunk size or ii) launch an error?

Reply:
Updated the code so that the size of the inner serial scan block (`element_values`) can be set by `m_policy.chunk_size()`. 
The chunk size should be power of two, and thus size-checking code is also added.

Reply:
Thanks!